documentation/src/content/docs/en/essentials/events.md

```diff
@@ -0,0 +1,283 @@
+
+```ts
+import { createEvent, sample } from "effector";
+import { debug } from "patronum";
+
+const firstTriggered = createEvent<void>();
+const secondTriggered = createEvent<void>();
+
+sample({
+  clock: firstTriggered,
+  target: secondTriggered,
+});
+
+debug(firstTriggered, secondTriggered);
+
+firstTriggered();
+// => [event] firstTriggered undefined
+// => [event] secondTriggered undefined
+```
+
+However, if your environment does not permit the addition of further dependencies, you may use the `watch` method with caution.
+
+```ts
+import { createEvent, sample } from "effector";
+
+const firstTriggered = createEvent<void>();
+const secondTriggered = createEvent<void>();
+
+sample({
+  clock: firstTriggered,
+  target: secondTriggered,
+});
+
+firstTriggered.watch(() => console.info("[event] firstTriggered"));
+secondTriggered.watch(() => console.info("[event] secondTriggered"));
+
+firstTriggered();
+// => [event] firstTriggered
+// => [event] secondTriggered
+```
+
+:::tip{title="Keep in mind"}
+The `watch` method neither handles nor reports exceptions, manages the completion of asynchronous operations, nor addresses data race issues.
+
+Its primary intended use is for short-term debugging and logging purposes.
+:::
+
+## Working with TypeScript (#typescript)
+
+When an event is invoked, TypeScript will verify that the type of the argument passed matches the type defined in the event, ensuring consistency and type safety within the code.
+
+This is also works for operators like [`sample`](/en/essentials/unit-composition) or `split`:
+
+```ts
+import { sample, createEvent } from "effector";
+
+const someHappened = createEvent<number>();
+const anotherHappened = createEvent<string>();
+
+sample({
+  // @ts-expect-error error:
+  // "clock should extend target type";
+  // targets: { clockType: number; targetType: string; }
+  clock: someHappened,
+  target: anotherHappened,
+});
+```
+
+## Working with multiple events (#working-with-events)
+
+Events in effector can be combined in various ways to create more complex logic. Let's look at the main approaches:
+
+### Creating derivative events (#derived-events)
+
+You can create a new event based on an existing one using the `map` method, which will be fired after original event:
+
+```ts
+import { createEvent, createStore } from "effector";
+
+const userClicked = createEvent<{ id: number; name: string }>();
+// Creating an event that will trigger only with the user's name
+const userNameSelected = userClicked.map(({ name }) => name);
+const $userName = createStore("").on(userNameSelected, (_, newName) => newName);
+
+// Usage
+userClicked({ id: 1, name: "John" });
+// userNameSelected will get 'John'
+```
+
+:::info{title="Derived events"}
+You cannot call derived events directly, but you can still subscribe to them for state changes or triggering other units.
+:::
+
+### Filtering events (#filtering-events)
```

Comment:
Filtering should happen through sample filter only, don't tell about `.filter` at all